python/pyspark/pandas/frame.py

```diff
@@ -8809,16 +8813,26 @@ def describe(self, percentiles: Optional[List[float]] = None) -> "DataFrame":
         max      3.0
         Name: numeric1, dtype: float64
         """
-        exprs = []
+        exprs_numeric = []
+        exprs_non_numeric = []
         column_labels = []
+        # For storing the name of non-numeric type columns
+        column_names = []
+        is_timestamp_types = []
         for label in self._internal.column_labels:
             psser = self._psser_for(label)
-            if isinstance(psser.spark.data_type, NumericType):
-                exprs.append(psser._dtype_op.nan_to_null(psser).spark.column)
+            spark_data_type = psser.spark.data_type
+            if isinstance(spark_data_type, NumericType):
+                exprs_numeric.append(psser._dtype_op.nan_to_null(psser).spark.column)
                 column_labels.append(label)
-
-        if len(exprs) == 0:
-            raise ValueError("Cannot describe a DataFrame without columns")
+            else:
+                exprs_non_numeric.append(psser.spark.column)
+                column_names.append(label[0])
+                # For the timestamp type we also need to retreive the `first` and `last`.
+                if isinstance(spark_data_type, TimestampType):
```

Comment:
We should also handle `TimestampNTZType`

Reply:
Got it, let me address it